EIPS/eip-2583.md

```diff
@@ -0,0 +1,295 @@
+  - `9000` for an ether-transfer
+  - `25000` for a new account-creation
+  - (`2300` would be refunded to the caller later)
+  - A total runtime `gas`-cost of `34K` gas would be required to handle this case.
+
+After this EIP: 
+
+- In addition to the `34K` an additional `penalty` would be added. 
+    - Possibly two, since the reference implementation does the balance-check twice, but it's unclear whether the compiled code would indeed perform the check twice. 
+- A total runtime `gas`-cost of `34K+penalty` (or `34K + 2 * penalty`) would be required to handle this case. 
+
+It can be argued that the extra penalty of `2-3K` gas can be considered marginal in relation to the other `34K` gas already required to handle this.
+
+## Test Cases
+
+The following cases need to be considered and tested: 
+
+- That during creation of a brand new contract, within the constructor, the `penalty` should not be applied for calls concerning the self-address. 
+- TBD: How the `penalty` is applied in the case of a contract which has performed a `selfdestruct` 
+  - a) previously in the same call-context, 
+  - b) previously in the same transaction, 
+  - c) previously in the same block,
+  For any variant of `EXTCODEHASH(destructed)`, `CALL(destructed)`, `CALLCODE(destructed)` etc. 
+- The effects on a `transaction` with `0` value going to a non-existent account.
+
+## Security Considerations
+
+See 'Backwards Compatibility'
+
+## Implementation
+
+Not yet available. 
+
+## Alternative variants
+
+### Alt 1: Insta-refunds
+
+Bump all trie accesses with `penalty`. `EXTCODEHASH` becomes `2700` instead of `700`. 
+- If a trie access hit an existing item, immediately refund penalty (`2K` )
+
+Upside:
+
+- This eliminates the 'shielded relay' attack
+
+Downside: 
+
+- This increases the up-front cost of many ops (CALL/EXTCODEHASH/EXTCODESIZE/STATICCALL/EXTCODESIZE etc)
+  - Which may break many contracts. 
+
+### Alt 2: Parent bail
+
+Use `penalty` as described, but if a child context goes OOG on the `penalty`, then the remainder is subtracted from the 
+parent context (recursively).
+
+Upside:
+
+- This eliminates the 'shielded relay' attack
+
+Downside: 
+
+- This breaks the current invariant that a child context is limited by whatever `gas` was allocated for it. 
+  - However, the invariant is not _totally_ thrown out, the new invariant becomes that it is limited to `gas + penalty`.
+- This can be seen as 'messy' -- since only _some_ types of OOG (penalties) becomes passed up the call chain, but not others, e.g. OOG due to trying
+  to allocate too much memory. There is a distinction, however: 
+  - Gas-costs which arise due to not-yet-consumed resources do not get passed to parent. For example: a huge allocation is not actually performed if there is insufficient gas. 
+  - Whereas gas-costs which arise due to already-consumed resources _do_ get passed to parent; in this case the penalty is paid post-facto for a trie iteration. 
```

Comment:
All of this should either move to discussion-to link or be restructured to fit into the Rationale section (the section for "why did we do X instead of Y").  EIPs are technical documents that define a specification, they aren't meant to be discussion platforms or a log of discussions on a topic.